lib/auth/auth_with_roles.go

```diff
@@ -2298,8 +2300,22 @@ func (a *ServerWithRoles) UpsertAuthServer(ctx context.Context, s types.Server)
 }
 
 func (a *ServerWithRoles) GetAuthServers() ([]types.Server, error) {
+	if a.scopedContext != nil {
+		ruleCtx := a.scopedContext.RuleContext()
+		if err := a.scopedContext.CheckerContext.RiskyUnpinnedDecision(a.CloseContext(), scopes.Root, func(checker *services.SplitAccessChecker) error {
```

Comment:
nit: I think it would be good to include justification comments explaining the use of risky-unpinned here, similar to how we do for reading CAs.  Something like:

```suggestion
		// for auth server reads we do not enforce scope pinning. this ensures that auths are readable for
		// all scoped identities regardless of their current scope pinning. this pattern should not
		// be used for any checks save essential global configuration reads that are necessary for basic
		// teleport functionality.
		if err := a.scopedContext.CheckerContext.RiskyUnpinnedDecision(a.CloseContext(), scopes.Root, func(checker *services.SplitAccessChecker) error {
```